backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -436,6 +465,19 @@ public Response updateEndpoint(@Context SecurityContext sec,
 
         endpointRepository.updateEndpoint(endpoint);
 
+        // Update the secrets in Sources.
+        if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
+            // In order to be able to update the secrets in Sources, we need to grab the IDs of these secrets from the
+            // database endpoint, since the client won't be sending those IDs.
+            var incomingEndpointProps = (SourcesSecretable) endpoint.getProperties();
+            var databaseEndpointProps = (SourcesSecretable) ep.getProperties();
```

Comment:
Unless I missed something, this could throw a `ClassCastException` if `endpoint` does not implement `SourcesSecretable`. It would be better to check if `endpoint instanceof SourcesSecretable` before doing this.

Reply:
Nice catch :fishing_pole_and_fish: ! I was just trying with `Webhook` and `Camel` endpoints and I forgot about that type check